src/core/annotation.js

```diff
@@ -1895,47 +1895,45 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
       );
     }
 
+    // The annotationStorage can be null if no change occured on the form
+    // but a checkbox could be already checked when it's open.
+
+    let value = null;
     if (annotationStorage) {
       const storageEntry = annotationStorage.get(this.data.id);
-      const value = storageEntry && storageEntry.value;
-      if (value === undefined) {
-        return super.getOperatorList(
-          evaluator,
-          task,
-          renderForms,
-          annotationStorage
-        );
-      }
+      value = (storageEntry && storageEntry.value) || null;
+    }
 
-      let appearance;
-      if (value) {
-        appearance = this.checkedAppearance;
+    if (value === null) {
+      if (this.data.checkBox) {
+        value = this.data.fieldValue && this.data.fieldValue !== "Off";
       } else {
-        appearance = this.uncheckedAppearance;
+        value = this.data.fieldValue === this.data.buttonValue;
       }
+    }
 
-      if (appearance) {
-        const savedAppearance = this.appearance;
-        this.appearance = appearance;
-        const operatorList = super.getOperatorList(
-          evaluator,
-          task,
-          renderForms,
-          annotationStorage
-        );
-        this.appearance = savedAppearance;
-        return operatorList;
-      }
+    let appearance;
+    if (value) {
+      appearance = this.checkedAppearance;
+    } else {
+      appearance = this.uncheckedAppearance;
+    }
```

Comment:
I see that this is pre-existing code, but how about simplifying it a bit when you're already touching it?
```javascript
const appearance = value ? this.checkedAppearance : this.uncheckedAppearance;
```